python/BioSimSpace/_SireWrappers/_system.py

```diff
@@ -2114,6 +2155,56 @@ def getRestraintAtoms(
 
         return indices
 
+    def containsAminoAcids(self, property_map={}):
+        """
+        Check whether the system contains nucleic acids.
+
+        Parameters
+        ----------
+        property_map : dict
+           A dictionary that maps system "properties" to their user defined
+           values. This allows the user to refer to properties with their
+           own naming scheme, e.g. { "charge" : "my-charge" }
+
+        Returns
+        -------
+
+        contains_aa : bool
+            True if the system contains amino acids, False otherwise.
+        """
+        string = "(resname " + ",".join(self.prot_res) + ")"
+        try:
+            search = self.search(string, property_map)
+        except:
+            search = []
+
+        return bool(search)
+
+    def containsNucleicAcids(self, property_map={}):
+        """
+        Check whether the system contains nucleic acids.
+
+        Parameters
+        ----------
+        property_map : dict
+           A dictionary that maps system "properties" to their user defined
+           values. This allows the user to refer to properties with their
+           own naming scheme, e.g. { "charge" : "my-charge" }
+
+        Returns
+        -------
+
+        contains_na : bool
+            True if the system contains amino acids, False otherwise.
+        """
+        string = "(resname " + ",".join(self.nucl_res) + ")"
+        try:
+            search = self.search(string, property_map)
+        except:
+            search = []
+
+        return bool(search)
```

Comment:
Is it possible to match these to the other similar search functions, e.g. have functions called `nNucleicAcids` and `getNucleicAcids`, etc. The first would give you the number of matches, the second returns the atoms matching the search, if any. (Could probably do the same for ions, too.)

Reply:
Sure - I've added these. I've skipped `getIons` for now because this seems a bit more complicated. The `ions` list only contains atom names, so this would have to be `getMonoatomicIons`, and I'd have to make sure that the atoms aren't part of molecules. Happy to try and implement something if you would like though·